components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/ThemeManagerHostObject.java

```diff
@@ -0,0 +1,437 @@
+            throws AppManagementException {
+
+        if (args == null || args.length != 2) {
+            handleException(
+                    "Invalid input parameters for getDeployedThemes.Expected parameters : tenantDomain,themeTypes");
+        }
+        String tenantDomain = (String) args[0];
+        NativeArray themeTypes = (NativeArray) args[1];
+        NativeObject themes = new NativeObject();
+
+        if (log.isDebugEnabled()) {
+            String msg = String.format("Get deployed themes for tenant :%s", tenantDomain);
+            log.debug(msg);
+        }
+        //check tenant theme dir exists
+        String path = getTenantThemePath(tenantDomain);
+        File dir = new File(path);
+        if (!dir.exists()) {
+            return themes;
+        }
+
+        for (Object themeType : themeTypes) {
+            String theme = (String) themeType;
+            path = getCustomThemePath(tenantDomain, (String) themeType);
+            themes.put(theme, themes, isThemeExists(path));
+        }
+        return themes;
+    }
+
+    /**
+     * Remove given deployed custom them for given tenant.
+     *
+     * @param cx
+     * @param thisObj
+     * @param args
+     * @param funObj
+     * @return true if success
+     * @throws AppManagementException
+     */
+    public static boolean jsFunction_deleteCustomTheme(Context cx, Scriptable thisObj, Object[] args, Function funObj)
+            throws AppManagementException {
+        if (args == null || args.length != 2) {
+            handleException(
+                    "Invalid input parameters for getDeployedThemes.Expected parameters : tenantDomain,themeType");
+        }
+        String tenantDomain = (String) args[0];
+        String themeType = (String) args[1];
+        return undeployTheme(tenantDomain, themeType);
+
+    }
+
+    /**
+     * Delete the theme directory in the file system based on given theme type.
+     *
+     * @param tenantDomain Tenant Domain
+     * @param themeType    Theme Type (Default ,<assetType> e.g webapp)
+     * @return true if success.
+     * @throws AppManagementException
+     */
+    private static boolean undeployTheme(String tenantDomain, String themeType) throws AppManagementException {
+        if (log.isDebugEnabled()) {
+            String msg = String.format("Delete custom theme:%1s of tenant : %2s", themeType, tenantDomain);
+            log.debug(msg);
+        }
+        // store/themes/<tenantDomain>
+        String tenantThemePath = getTenantThemePath(tenantDomain);
+        // store/themes/<tenantDomain>/extensions/assets
+        String extAssetsPath = getExtThemePath(tenantDomain);
+        // store/themes/<tenantDomain>/themes
+        String defaultThemePath = tenantThemePath + File.separator + "themes";
+        // store/themes/<tenantDomain>/extensions
+        String extPath = tenantThemePath + File.separator + "extensions";
+
+        File dir = new File(tenantThemePath);
+        if (!dir.exists()) {
```

Comment:
Use [Files.exists(...)](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#exists(java.nio.file.Path,%20java.nio.file.LinkOption...)) instead of this.